### PR TITLE
Add support for NSX-T Edge Gateway BGP Neighbor configuration

### PR DESCRIPTION
**Pull request by Didainius**
This PR continues on the track of PRs #480 and #488 and completes coverage of NSX-T Edge Gateway BGP Configuration by adding Neighbor configuration support.

Add support for managing NSX-T Edge Gateway BGP Neighbor. It is done by adding types `EdgeBgpNeighbor` and
  `types.EdgeBgpNeighbor` with functions `CreateBgpNeighbor`, `GetAllBgpNeighbors`,
  `GetBgpNeighborByIp`, `GetBgpNeighborById`, `Update` and `Delete`

Note. VCD 10.2 versions have a problem - they do not return a new entity ID when it is created. Function `CreateBgpNeighbor` handles this internally by waiting for task completion and checking if ID was returned (it is returned in 10.3). If the ID is not returned - the function will lookup the entity Neighbor IP address. This should be safe (as opposed to lookup by name in `CreateBgpIpPrefixList`) as the uniqueness of this IP is enforced by API

### CODE REVIEW

File: types/v56/constants.go [hunk 1]
```diff
@@ -375,6 +375,7 @@ const (
 	OpenApiEndpointSecurityTags                       = "securityTags"
 	OpenApiEndpointNsxtRouteAdvertisement             = "edgeGateways/%s/routing/advertisement"
 
+	OpenApiEndpointEdgeBgpNeighbor = "edgeGateways/%s/routing/bgp/neighbors/" // '%s' is NSX-T Edge gateway ID
```
Comment (lvirbalas) [inline types/v56/constants.go:378]:
```suggestion
	OpenApiEndpointEdgeBgpNeighbor = "edgeGateways/%s/routing/bgp/neighbors/" // '%s' is NSX-T Edge Gateway ID
```

File: types/v56/nsxt_types.go [hunk 2]
```diff
@@ -1254,3 +1254,74 @@ type RouteAdvertisement struct {
...
+	// direction. It is the reference to the prefix list, indicating which routes to filter for IN
+	// direction. Not specifying a value will be treated as "no IN route filters".
+	InRoutesFilterRef *OpenApiReference `json:"inRoutesFilterRef,omitempty"`
+
+	// OutRoutesFilterRef specifies route filtering configuration for the BGP neighbor in 'OUT'
+	// direction. It is the reference to the prefix list, indicating which routes to filter for OUT
+	// direction. Not specifying a value will be treated as "no OUT route filters".
+	OutRoutesFilterRef *OpenApiReference `json:"outRoutesFilterRef,omitempty"`
+
+	// Specifies the BFD  (Bidirectional Forwarding Detection) configuration for failure detection. Not specifying a value
```
Comment (lvirbalas) [inline types/v56/nsxt_types.go:1311]:
```suggestion
	// Specifies the BFD (Bidirectional Forwarding Detection) configuration for failure detection. Not specifying a value
```

**Review by lvirbalas — APPROVED** [review]
Few polishes and LGTM!

**Review by adambarreiro — APPROVED** [review]
LGTM! Will run tests once I have an environment ready.

File: types/v56/nsxt_types.go (hunk 2 above)
Comment (Didainius) [reply types/v56/nsxt_types.go:1339]: Fixed.

File: types/v56/constants.go (hunk 1 above)
Comment (Didainius) [reply types/v56/constants.go]: Fixed.

**Pull request merged**